common/flatpak-dir.c

```diff
@@ -5935,6 +5936,19 @@ flatpak_dir_pull (FlatpakDir                           *self,
                                      error))
     goto out;
 
+  /* Work around a libostree bug where the pull may succeed but the pulled
+   * commit will be incomplete by preemptively marking the commit partial.
+   * Note this has to be done before ostree_repo_prepare_transaction() so we
+   * aren't checking the staging dir for the commit.
+   * https://github.com/flatpak/flatpak/issues/3479
+   * https://github.com/ostreedev/ostree/pull/2549
+   */
+  if (!ostree_repo_has_object (repo, OSTREE_OBJECT_TYPE_COMMIT, rev, &have_commit, NULL, NULL))
+    g_warning ("Encountered error checking for commit object %s", rev);
+  else if (!have_commit &&
+           !ostree_repo_mark_commit_partial (repo, rev, TRUE, NULL))
+    g_warning ("Encountered error marking commit partial: %s", rev);
```

Comment:
It might be better to use the GError from those functions:

```suggestion
  {
    g_autoptr(GError) local_error = NULL;

    if (!ostree_repo_has_object (repo, OSTREE_OBJECT_TYPE_COMMIT, rev, &have_commit, NULL, &local_error))
      g_warning ("Encountered error checking for commit object %s: %s", rev, local_error->message);
    else if (!have_commit &&
             !ostree_repo_mark_commit_partial (repo, rev, TRUE, &local_error))
      g_warning ("Encountered error marking commit partial: %s: %s", rev, local_error->message);
  }
```

Reply:
Sure we can do this. I didn't because it's only a temporary patch until we can require the patched libostree, and the error seemed unlikely, but it is essentially costless to use the error pointer.